cmd/k8s-operator/tsrecorder.go

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	xslices "golang.org/x/exp/slices"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"tailscale.com/client/tailscale"
+	tsoperator "tailscale.com/k8s-operator"
+	tsapi "tailscale.com/k8s-operator/apis/v1alpha1"
+	"tailscale.com/tailcfg"
+	"tailscale.com/tstime"
+	"tailscale.com/util/clientmetric"
+	"tailscale.com/util/set"
+)
+
+const (
+	reasonTSRecorderCreationFailed = "TSRecorderCreationFailed"
+	reasonTSRecorderCreated        = "TSRecorderCreated"
+	reasonTSRecorderInvalid        = "TSRecorderInvalid"
+
+	currentProfileKey = "_current-profile"
+)
+
+var gaugeTSRecorderResources = clientmetric.NewGauge("k8s_tsrecorder_resources")
```

Comment:
Nit: since you added these lines, I've moved all the metric names to https://github.com/tailscale/tailscale/blob/a228d77f8620cb6ba693e28cda48705c8418f7e7/kube/kubetypes/metrics.go#L 15-L22, could we add this one there too?